x-pack/plugins/ml/public/application/datavisualizer/index_based/index_data_visualizer.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, Fragment, useEffect, useState } from 'react';
+import { useMlKibana, useTimefilter } from '../../contexts/kibana';
+import { NavigationMenu } from '../../components/navigation_menu';
+import { HelpMenu } from '../../components/help_menu';
+import type { IndexDataVisualizerViewProps } from '../../../../../data_visualizer/public';
+export const IndexDataVisualizerPage: FC = () => {
+  useTimefilter({ timeRangeSelector: false, autoRefreshSelector: false });
+  const {
+    services: { docLinks, dataVisualizer },
+  } = useMlKibana();
+  const [
+    IndexDataVisualizer,
+    setIndexDataVisualizer,
+  ] = useState<FC<IndexDataVisualizerViewProps> | null>(null);
+
+  useEffect(() => {
+    if (dataVisualizer !== undefined) {
+      const { getIndexDataVisualizerComponent } = dataVisualizer;
+      getIndexDataVisualizerComponent().then(setIndexDataVisualizer);
+    }
+  }, []);
+  return IndexDataVisualizer ? (
```

Comment:
not for this PR, but it would be good to switch this and the file data viz to use `React.lazy` rather than these cost lazy loaders.
file data viz has `FileDataVisualizerWrapper` to do this when embedding in `home`, but I didn't change how ML was doing it as I intended to change it after this PR was in.

Reply:
I'll add this to the list for the next follow up 👍